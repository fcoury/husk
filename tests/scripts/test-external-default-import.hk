// Test importing default exports from external packages
// This test only verifies the transpilation output, not runtime behavior

// These imports should generate: import name from 'package'
use external::test_package_a::default;
use external::test_package_b::default;

fn main() {
    // Just verify the imports are generated correctly
    println("Default import test");
}