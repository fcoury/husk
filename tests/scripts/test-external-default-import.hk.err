error: 4:1 - External package 'external::test_package_a::default' cannot be used in interpreter mode. Use the transpiler to convert to JavaScript for external package support.
use external::test_package_a::default;
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
