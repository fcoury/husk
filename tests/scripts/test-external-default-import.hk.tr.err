node:internal/modules/package_json_reader:255
  throw new ERR_MODULE_NOT_FOUND(packageName, fileURLToPath(base), null);
        ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find package 'test_package_a' imported from /Users/fcoury/code/husk-build-command/[eval1]
    at Object.getPackageJSONURL (node:internal/modules/package_json_reader:255:9)
    at packageResolve (node:internal/modules/esm/resolve:767:81)
    at moduleResolve (node:internal/modules/esm/resolve:853:18)
    at defaultResolve (node:internal/modules/esm/resolve:983:11)
    at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:801:12)
    at #cachedDefaultResolve (node:internal/modules/esm/loader:725:25)
    at ModuleLoader.resolve (node:internal/modules/esm/loader:708:38)
    at ModuleLoader.getModuleJobForImport (node:internal/modules/esm/loader:309:38)
    at #link (node:internal/modules/esm/module_job:201:49)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5) {
  code: 'ERR_MODULE_NOT_FOUND'
}

Node.js v24.2.0
