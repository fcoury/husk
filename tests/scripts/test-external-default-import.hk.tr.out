import test_package_a from 'test_package_a';
import test_package_b from 'test_package_b';
function main() {
  void (println("Default import test"));
}